service/storage_service.cc

```diff
@@ -4664,6 +4664,41 @@ future<> storage_service::wait_for_topology_not_busy() {
     }
 }
 
+future<locator::effective_replication_map_ptr> storage_service::get_erm_when_table_has_no_migrations(sstring ks_name, sstring table_name) {
+    lw_shared_ptr<replica::table> t;
+    try {
+        t = _db.local().find_column_family(ks_name, table_name).shared_from_this();
+    } catch (replica::no_such_column_family& e) {
+        throw std::runtime_error(format("Table {}.{} does not exist", ks_name, table_name));
```

Comment:
nit: what's wrong with just letting the `replica::no_such_column_family` exception sip through?

Reply:
A more elegant implementation can use `get_tables_metadata().get_table_id_if_exists()` and then `get_table()` if the id is not false, else throw the exception.

Reply:
BTW, throwing `replica::no_such_column_family` would be better a choice IMO, since some callers would want to tolerate it (e.g. when taking a snapshot, and even truncate could in theory ignore a dropped table if it's dropped in parallel)

Reply:
My knowledgeof group0 code is a bit rotten, but I don't see how a reference to erm prevents pending migrations from stacking up in the tablet state machine. @tgrabiec ?

Reply:
Migration executes a topology barrier in the first stage, so holding erm where no tablet is in transition will prevent migrations from advancing to any stage which has any effect on tablet storage. 

user-requested repair currently uses this method.

But we're abandoning this method for truncate, because we want truncate to also serialize with repair, and we don't want truncate to block indefinitely behind load balancing. 

Repair will also switch to being scheduled by the tablet scheduler, there's ongoing work by Asias for that.

Reply:
Closing this conversation as it relates to the abandoned design.